### PR TITLE
LPS-152515 search-experiences-web: Add js unit tests for JSON autocompletion

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-152515 - Add js-unit test for JSON auto-completion feature in LPS-148749

Test cases are confined to the CodeMirrorEditor and use the `getCodeMirrorHints` function! There were some small updates made to `CodeMirrorEditor.js` (export function, remove usage of `.at()` function, change `innerText` to `textContent` assignment) to get the tests to pass.

Test coverage includes:
 - getting simple hints
 - hints for matched string
 - hints for enum
 - hints with languageId as a ref
 - hints when using `allOf`
 - hints when using `additionalProperties`
 - display the hint with name/type(s)
 - autocompletion for different types (array, object, string, `aggs`)

Let me know what you think!

### CODE REVIEW

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/shared/CodeMirrorEditor.js [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+		...props,
+	});
+
+	codeMirror.execCommand('goLineEnd');
+
+	return codeMirror;
+}
+
+describe('CodeMirrorEditor', () => {
+	it('gets the correct hints', () => {
```
Comment (thektan) [inline modules/dxp/apps/search-experiences/search-experiences-web/test/js/shared/CodeMirrorEditor.js:56]: Can we make this more specific so it's more helpful what's being tested like `gets the hints for the root level of an object`?

Comment (oliv-yu) [reply modules/dxp/apps/search-experiences/search-experiences-web/test/js/shared/CodeMirrorEditor.js]: Ok! I used that phrase but tweaked a word or two 😅

File: modules/dxp/apps/search-experiences/search-experiences-web/test/js/shared/CodeMirrorEditor.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	es_ES: 'Spanish (Spain)',
+	fr_FR: 'French (France)',
+};
+
+global.document.body.createTextRange = () => {
+	return {
+		getBoundingClientRect: () => {},
+		getClientRects: () => [],
+	};
+};
```
[the remark this answers is not in the document]
Comment (oliv-yu) [reply modules/dxp/apps/search-experiences/search-experiences-web/test/js/shared/CodeMirrorEditor.js:40]: Added! I tried to keep the comment similar to comments in `EditSXPBlueprintForm.js` test.

**Comment by thektan** [comment]
@oliv-yu Thanks for those changes! 💯

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:forward

**Comment by thektan** [comment]
ci:forward

**Pull request closed without merging**